### PR TITLE
webgpu: add fusedBatchMatmul

### PR DESCRIPTION
**Pull request by qjia7**
DEV
PERF

This patch adds fusedBatchMatmul operator and optimizes fusedConv2d using it
when conditions are satisfied.

This patch also removes MatMulProgram since it's same with MatMulPackedProgram
when WEBGPU_MATMUL_WORK_PER_THREAD is 1.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4577)

### CODE REVIEW

**Comment by qjia7** [comment]
@lina128 @kainino0x Please take a look, thanks.
In this patch, I haven't done the modularization since it will make the changes more complicated.  
@haoyunfeix will do the whole modularization work.

File: tfjs-backend-webgpu/src/kernels/matmul_packed_webgpu.ts [hunk 1]
```diff
@@ -193,7 +197,37 @@ export class MatMulPackedProgram implements WebGPUProgram {
             B[batch * ${batchBSize} + col * dimInner + row] : 0`;
     }
 
+    let activationSnippet = '', applyActivationSnippet = '';
```
Comment (kainino0x) [inline tfjs-backend-webgpu/src/kernels/matmul_packed_webgpu.ts:200]: Any chance of sharing the duplicated (or mostly duplicated?) block here between the two implementations?

**Review by kainino0x — APPROVED** [review]
LGTM

**Review by lina128 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4577)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @qjia7)

___
*[tfjs-backend-webgpu/src/flags_webgpu.ts, line 37 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4577#-MRXEqya9ctEzsfSvvpG:-MRXEqya9ctEzsfSvvpH:b-6e0nzo) ([raw file](https://github.com/tensorflow/tfjs/blob/3d44cc00b67481a7f65d83a0b39b8172b1543675/tfjs-backend-webgpu/src/flags_webgpu.ts#L37)):*
> ```XML
> 
> /**
>  * Whether to use conv2d_naive.
> ```

Please add a description of conv2d_naive.

**Review by lina128 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4577)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @qjia7)

___
*[tfjs-backend-webgpu/src/flags_webgpu.ts, line 37 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4577#-MRXEqya9ctEzsfSvvpG:-MRXcFxVAB16EVWPf0Qh:bebeqqf) ([raw file](https://github.com/tensorflow/tfjs/blob/3d44cc00b67481a7f65d83a0b39b8172b1543675/tfjs-backend-webgpu/src/flags_webgpu.ts#L37)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Please add a description of conv2d_naive.
</blockquote></details>

Thank you! LGTM.

File: tfjs-backend-webgpu/src/kernels/matmul_packed_webgpu.ts (hunk 1 above)
Comment (qjia7) [reply tfjs-backend-webgpu/src/kernels/matmul_packed_webgpu.ts:200]: Yes, it's a little noising that lots of code between them looks very similar and the include the code in conv2d. I will consider how to refactor this part and do it in a separate PR. Thanks.

**Pull request merged**